packages/syft/src/syft/custom_worker/builder_k8s.py

```diff
@@ -0,0 +1,318 @@
+# stdlib
+from hashlib import sha256
+import os
+from pathlib import Path
+from typing import Dict
+from typing import Optional
+
+# third party
+import kr8s
+from kr8s.objects import APIObject
+from kr8s.objects import ConfigMap
+from kr8s.objects import Job
+
+# relative
+from .builder_types import BuilderBase
+from .builder_types import ImageBuildResult
+from .builder_types import ImagePushResult
+
+__all__ = ["KubernetesBuilder"]
+
+JOB_COMPLETION_TTL = 60
+
+BUILD_OUTPUT_PVC = "worker-builds"
+KUBERNETES_NAMESPACE = os.getenv("K8S_NAMESPACE", "syft")
+
+
+class InvalidImageDigest(Exception):
+    pass
+
+
+class KubernetesBuilder(BuilderBase):
+    def __init__(self):
+        self.client = kr8s.api(namespace=KUBERNETES_NAMESPACE)
+
+    def build_image(
+        self,
+        tag: str,
+        dockerfile: str = None,
+        dockerfile_path: Path = None,
+        buildargs: Optional[dict] = None,
+        **kwargs,
+    ) -> ImageBuildResult:
+        job_id = self._new_job_id(tag)
+
+        if dockerfile:
+            pass
+        elif dockerfile_path:
+            dockerfile = dockerfile_path.read_text()
+
+        # Create a ConfigMap with the Dockerfile
+        config = self._create_build_config(job_id, dockerfile)
+        config.refresh()
+
+        # Create and start the job
+        job = self._create_kaniko_build_job(
+            job_id=job_id,
+            tag=tag,
+            job_config=config,
+            build_args=buildargs,
+        )
+
+        try:
+            # wait for job to complete/fail
+            job.wait(["condition=Complete", "condition=Failed"])
+
+            # TODO: check job status, raise with logs
+
+            image_digest = self._get_image_digest(job)
+            if not image_digest:
+                raise InvalidImageDigest("Did not get any image digest from the job")
+
+            # get logs
+            logs = self._get_logs(job)
+        except Exception:
+            raise
```

Comment:
MAYBE: We could return a SyftError here in the except block, that is automatically handled in worker_image_service.py build method

Reply:
maybe , we could also add 
```python3
# Create and start the job
        job = self._create_kaniko_build_job(
            job_id=job_id,
            tag=tag,
            job_config=config,
            build_args=buildargs,
        )
```
as part of the try catch block